### ISSUE TITLE
Add editorconfig support in Razor

### ISSUE THREAD

**Issue by allisonchou** [dotnet/roslyn#85310]
Tracking issue on the Razor side for adding editorconfig support in Razor.
Will likely need to discuss with Platform since the bulk of the work may need to be done on their side.

See 1) in: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1388770/ for original VSTS issue.

----

There are a few different aspects to .editorconfig support, with different statuses and priorities:

**Current status:**

1. Support for `indent_size`, `indent_style` etc. generic formatting properties, as per the original linked issue above
	* Blocked by the Visual Studio editor, tracked internally by https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1408493
1. Support for start C# editor config controls in document formatting, eg code style
	* Not supported, is possible with bigger architectural changes like https://github.com/dotnet/razor/issues/6919
1. Support for razor specific editorconfig controls for formatting and other behaviours
	* Tracked by https://github.com/dotnet/razor/issues/8413

**Comment by sportfloh** [comment]
Hi,
what is the status on this issue? This issue has been opened 4+ years ago.
I have to work with C#/Blazor/Razor since last year and I'm extremely annoyed, why such basic things as formatting are not working.
I use VSCode + C# DevKit + Exitorconfig extension
Can this please be prioritized, so that this basic functionality will be available soon?

Related issues:
- https://github.com/dotnet/razor/issues/5607
- https://github.com/dotnet/sdk/issues/39817
- https://github.com/dotnet/sdk/issues/49347
- https://github.com/dotnet/format/issues/153

Thx and cheers!

**Comment by sportfloh** [comment]
@ davidwengier @ JoeRobich can you please comment and prioritize this? Thanks!

**Comment by davidwengier** [comment]
@sportfloh No updates to share I'm afraid. This is an extremely complex piece of work, with many dependencies, and we do not have the resources to work on it at the moment.

**Comment by Kebechet** [comment]
> [@sportfloh](https://github.com/sportfloh) No updates to share I'm afraid. This is an extremely complex piece of work, with many dependencies, and we do not have the resources to work on it at the moment.

"Claude fix issue, dont make mistakes" 😅

**Comment by icnocop** [comment]
It seems a global analyzer config file would work-around this issue, but the configurations are not file extension specific (`*.razor`) - they will apply to all compiled source files.

I added the file `.globalconfig` to the root of my repo, next to `.editorconfig`.
I only tested it to suppress a message in the IDE.

```
# https://learn.microsoft.com/dotnet/fundamentals/code-analysis/configuration-files#global-analyzerconfig
is_global = true
dotnet_diagnostic.XX1000.severity = none
```

SDK-style projects will automatically discover this file.

**Comment by sportfloh** [comment]
> [@sportfloh](https://github.com/sportfloh) No updates to share I'm afraid. This is an extremely complex piece of work, with many dependencies, and we do not have the resources to work on it at the moment.

@davidwengier thank you for your response! Let me know if I can help to implement this feature.
Cheers!

**Comment by davidwengier** [comment]
A note for anyone watching this issue, https://github.com/dotnet/razor/issues/5607 was recently fixed, and will appear in v18.11, which is the next major Insiders release. With that we will consider the C# formatting options, like spacing and newline rules around parentheses and braces, defined in `[*.razor]` blocks in your `.editorconfig` files. So if you have a `[*.cs]` with all of your rules right now, you may want to update it to `[*.{cs,razor,cshtml}]` in preparation. Not all of the options will work right now, due to a combination of factors, but this at least puts us on the right path. Once you have the update, feel free to log a new issue for any specific options that don't work.

**Comment by davidwengier** [comment]
With https://github.com/dotnet/roslyn/pull/85128 merged, formatting options are now applied for code actions too, which is currently everything we can do. For diagnostics filtering we could add support to VS and VS Code, however it wouldn't affect command line builds, so feels pretty useless to me. In particular for filtering, or controlling the severity of RZXXX diagnostics on command line builds, we'd need support from Roslyn (probably https://github.com/dotnet/roslyn/issues/31916). So I'm not closing this issue, but it will remain marked as external, but I would recommend for any non-diagnostics related editorconfig features or bugs, new issues are logged.
